pyproject.toml

```
@@ -122,7 +122,7 @@ docs = [
     "Jinja2>=3.1.3",
     "tomli; python_version < '3.11'",
     "sphinxcontrib-globalsubs >= 0.1.1",
-    "matplotlib>=3.9.1",  # https://github.com/matplotlib/matplotlib/issues/28234
+    "matplotlib==3.9.1",  # https://github.com/matplotlib/matplotlib/issues/28234 # TODO: switch to >=3.9.2 when available (3.9.1 is yanked)
```

If 3.9.1 is yanked, we shouldn't do a `==` pin to it? Am I missing something?
```suggestion
    "matplotlib!=3.9.1",  # https://github.com/matplotlib/matplotlib/issues/28234 # TODO: switch to >=3.9.2 when available (3.9.1 is yanked)
```

---

Our docs really does require >=3.9.1 at the moment, and because it's yanked but no newer version is available yet, the only way to get it is to require exactly this version.

---

Ah, https://github.com/matplotlib/matplotlib/issues/28234 ... Well, that's a headache. Feels wrong to pin to something that is yanked. Let me think a little.

---

I think matplotlib 3.9.1 still available from conda-forge (https://anaconda.org/conda-forge/matplotlib)? So, theoretically, we could keep this pin untouched and install matplotlib from conda-forge instead for RTD?

https://github.com/astropy/astropy/blob/384e90b5132e386faaf168fdb1bda100d3eb26ef/.readthedocs.yaml#L13-L14

https://github.com/astropy/astropy/blob/384e90b5132e386faaf168fdb1bda100d3eb26ef/docs/rtd_environment.yaml#L5

---

Hopefully this situation is short-lived enough upstream that we don't have to worry about users trying to build docs locally?

---

I don't feel strongly about it but yes, it seems that the situation will be resolved upstream in a couple days if I understand correctly. Feel free to push to this branch, I'll be AFK for the rest of the day !

---

It is only yanked because there was no other way to satisfy:

1. do not ship windows wheels that segfault
2. do not cause windows machines that are not set up to compile c/c++ try to install Matplotlib from source 

If this pinning only runs on linux, there is no problem.